controllers/utils/events.go

```diff
@@ -0,0 +1,101 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package utils
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	policyv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	PolicyFmtStr              = "policy: %s/%s"
+	PolicyClusterScopedFmtStr = "policy: %s"
+)
+
+// ComplianceEventSender handles sending policy template status events in the correct format.
+type ComplianceEventSender struct {
+	ClusterNamespace string
+	InstanceName     string
+	ClientSet        *kubernetes.Clientset
+	ControllerName   string
+}
+
+// SendEvent will send a policy template status message update synchronously as opposed to EventRecorder
+// sending events in the background asynchronously.
+func (c *ComplianceEventSender) SendEvent(
+	ctx context.Context,
+	instance client.Object,
+	owner metav1.OwnerReference,
+	reason string,
+	msg string,
+	compliance policyv1.ComplianceState,
+) error {
+	msg = string(compliance) + "; " + msg
```

Comment:
The issue here is that this will lead to duplicate compliance prefixes. See my comment here:
https://github.com/open-cluster-management-io/governance-policy-framework-addon/pull/76#discussion_r1304334436

Reply:
Maybe fixed in the new commit 😆